src/pybel/io/mongodb.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+"""An exporter from PyBEL graphs to a local Mongo database."""
+
+import logging
+from copy import deepcopy
+from typing import Any, List, Mapping, Optional, Tuple
+
+from pymongo import MongoClient
+from pymongo.collection import Collection
+
+from pybel import BELGraph, to_sbel
+from pybel.dsl import Entity, Variant
+from pybel.schema import is_valid_edge, is_valid_node
+
+__all__ = [
+    '_entity_to_dict',
+    '_rm_mongo_keys',
+    'to_mongodb',
+    'find_nodes',
+    'get_edges_from_node',
+    'get_edges_from_criteria'
+]
+
+logger = logging.getLogger(__name__)
+
+
+def _entity_to_dict(entity: Entity) -> Mapping[str, Any]:
+    """Input a pybel Entity and return a dict representing it."""
+    new_node = dict(entity)
+    if new_node['function'] in ['Complex', 'Composite']:
+        new_node['members'] = list(map(dict, new_node['members']))
+    return new_node
+
+
+def to_mongodb(
+    graph: BELGraph,
+    db_name: str, collection_name: str,
+    client: Optional[MongoClient] = None
+) -> Collection:
+    """Export the given BELGraph to a MongoDB.
+
+    In order to use this function, MongoDB must already be running locally.
+
+    :param graph: the graph to be exported
+    :param db_name: the name of the MongoDB to which the graph should be exported
+    :param collection_name: the name of the collection within the MongoDB where the graph will be stored.
+    :return: the collection that now stores the graph
+    """
+    if client is None:
+        client = MongoClient()
+    # Access (or create) the specified database and collection
+    db = client[db_name]
+    collection = db[collection_name]
+    # If a collection with the same name already exists, drop its contents
```

Comment:
why? what if someone uploads one graph, then later wants to upload another graph? or should the collection correspond to the graph?

Reply:
My thinking was that each collection would correspond to one graph.

Reply:
is it possible to query over all collections simultaneously?

Reply:
I'm not sure, but I don't think so

Reply:
Did you get a chance to look into this?

Reply:
I could write a function to query over all collections, but it would just be a loop, not simultaneous